services/user/XAdmin/ui/wasm/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+#[allow(warnings)]
+mod bindings;
+use std::{io::Cursor, str::FromStr};
+
+use bindings::*;
+
+use exports::psibase::x_admin::boot::{Guest as Boot, *};
+use exports::psibase::x_admin::packages::Guest as Packages;
+use exports::psibase::x_admin::util::Guest as Util;
+use psibase::fracpack::{Pack, Unpack};
+use psibase::*;
+
+const DEBUG_PRINT: bool = false;
+
+fn parse_public_key_pem(public_key_pem: Option<String>) -> Result<Option<AnyPublicKey>, String> {
+    public_key_pem
+        .map(|k| -> Result<AnyPublicKey, String> {
+            let data = pem::parse(k.trim()).map_err(|e| e.to_string())?;
+
+            if data.tag() != "PUBLIC KEY" {
+                return Err("Invalid public key".to_string());
+            }
+
+            spki::SubjectPublicKeyInfoRef::try_from(data.contents())
+                .map_err(|e| format!("Invalid SPKI format: {}", e))?;
+
+            Ok(AnyPublicKey {
+                key: crate::Claim {
+                    service: AccountNumber::from_str("verify-sig").unwrap(),
+                    rawData: data.contents().to_vec().into(),
+                },
+            })
+        })
+        .transpose()
+}
+
+struct XAdminPlugin;
+
+impl Boot for XAdminPlugin {
+    fn boot_transactions(
+        producer: String,
+        js_services: Vec<Code>,
+        block_signing_key: Option<Pem>,
+        tx_signing_key: Option<Pem>,
+        compression_level: u32,
+    ) -> Result<(Tx, Vec<Tx>), String> {
```

Comment:
The signature is somewhat misleading. The first `Tx` is actually a packed `Vec<Tx>`.